### PR TITLE
Feat/prod3 254

### PR DESCRIPTION
**Pull request by nathielymm**
# Description

- Adds checkbox to allow marketing emails to ticket donation done page
- reactors donation done cause page -> contribution done page and ticket donation done page
- removes unused donation done  page
- updates shared lib

to do later: updates "from: donation-done-cause-page" on events (It is necessary to align with teams so as not to lose track of events) 

## Does this pull request close any open issues?

- Put here the issue that must be closed with this PR
- eg: closes [PROD3-254](https://ribon.atlassian.net/browse/PROD3-254)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test A

### CODE REVIEW

File: src/pages/donations/TicketDonationDonePage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  const integrationId = useIntegrationId();
+
+  const quantityOfDonationsToShowEmailCheckbox = 3;
+  const firstDonation = 1;
+
+  const { refetch } = useFirstAccessToIntegration(integrationId);
+
+  const shouldShowEmailCheckbox = useCallback(
+    () =>
+      (Number(userStatistics?.totalTickets) <=
```
Comment (thaisribon) [inline src/pages/donations/TicketDonationDonePage/index.tsx:70]: I think here if we put if(userStatistics && config) before all we will avoid checkbox from appearing and then disappear

`  const shouldShowEmailCheckbox = useCallback(() => {
    if (userStatistics && config) {
      return (
        (Number(userStatistics?.totalTickets) <=
          quantityOfDonationsToShowEmailCheckbox ||
          Number(userStatistics?.totalTickets) %
            quantityOfDonationsToShowEmailCheckbox ===
            0 ||
          Number(userStatistics?.totalTickets) === firstDonation) &&
        !config?.allowedEmailMarketing
      );
    } else {
      return false;
    }
  }, [userStatistics, config]);`

File: src/components/atomics/sections/IconsAroundImage/styles.ts [hunk 2]
```diff
@@ -29,7 +33,9 @@ export const Icon = styled.img<{ position: number; isStatic: boolean }>`
     }
 
     return `
-      animation: ${animationName} 4s cubic-bezier(0.1, 0.83, 1, 0.3) infinite;
+      animation: ${animationName} 4s cubic-bezier(0.1, 0.83, 1, 0.3) ${
```
Comment (thaisribon) [inline src/components/atomics/sections/IconsAroundImage/styles.ts:36]: why is this necessary? this way the icons are looking like this
Captura de Tela 2023-11-28 às 10 51 04

File: src/pages/donations/TicketDonationDonePage/index.tsx (hunk 1 above)
Comment (nathielymm) [reply src/pages/donations/TicketDonationDonePage/index.tsx]: Fixed it!

**Review by thaisribon — APPROVED** [review]
